### ISSUE TITLE
isEmptyOr [feature]

### ISSUE THREAD

**Issue by Flemin-Adam-zz** [ramda/ramda#2665]
A nice to have feature where we can return a default value if isEmpty returns false just like propOr

**Comment by CrossEye** [comment]
Could you describe some examples?  I don't think I've ever needed anything like this, but I could just be missing it.

**Comment by ben-eb** [comment]
Do you mean that if `isEmpty` returns `true` you want to return a default value?

```js
const isEmptyOr = value => ifElse(isEmpty, always(value), T)

isEmptyOr('A')('B') // => true
isEmptyOr('A')('')  // => 'A'
```

**Comment by Flemin-Adam** [comment]
@ben-eb Exactly. Something similar is ISNULL()(MSSQL) or IFNULL()(MYSQL). We can specify a value as the second parameter if the isEmpty returns true. Also, will this be like a "Good First Issue"? I would like to contribute given I have never contributed.

**Comment by foxbunny** [comment]
@ben-eb I think it's more along the lines of 

```
const valueOr = R.curry((def, x) => R.isNil(x) ? def : x)

valueOr('SAFE', 'A')  // => 'A'
valueOr('SAFE', null)  // => 'SAFE'
```

**Comment by foxbunny** [comment]
I'm super confused by this. I originally thought you meant "if `isEmpty()` returns true" but now I see it's the opposite. `propOr()` doesn't work like that.

**Comment by Flemin-Adam** [comment]
Ok. Maybe I'm not good with descriptions. I was thinking of a usage like this:

```
let a = '';
if(isEmptyOr(a,0)=== 0) doSomething();
```

**Comment by foxbunny** [comment]
See, that's what confuses me. `0` is simply an alias for `true`. I don't think that's a very valuable feature. You could as easily have said:

```
let a = '';
if(isEmpty(a)) doSomething();
```

**Comment by Flemin-Adam** [comment]
I see your point now. I agree. But it would probably be still useful in other scenarios. 

```
let a = '';
let b = 100/isEmptyOr(a,1)
```

**Comment by CrossEye** [comment]
I'm still not convinced of the usefulness and general applicability of this proposal.

First of all, a Ramda version would probably look more like this:

```js
const isEmptyOr = curry((dflt, val) => isEmpty(val) ? dflt : val)
```

While there's nothing wrong with the idea, and it would certainly be welcome in the Cookbook, I haven't seen anything that makes it a candidate for the library proper.  Part of the problem is that as far as I can tell the generic nature of this would obscure any likely usage, and there wouldn't be any likely call for the generic version itself.

But I could still be missing the use-cases.  Can you suggest others?
